aten/src/ATen/native/quantized/affine_quantizer.cpp

```diff
@@ -314,6 +395,25 @@ CAFFE2_API float dequantize_val(double scale, int64_t zero_point, T value) {
 }
 #endif // USE_FBGEMM
 
+/*
+* Quantize value based on the following equation
+* Qx = (Xf - bias) * inv_scale
+* bias and scale values are in floating point to enable higher precision computation
+* of the quantized value.
+*/
+template <typename T>
+T quantize_val_float_qparams(float scale, float bias, float value) {
+  int64_t qvalue;
+
+  // TODO make sure qmax and qmin for dtypes other than int8, uint8 is correctly defined.
+  constexpr int64_t qmin = std::numeric_limits<typename T::underlying>::min();
+  constexpr int64_t qmax = std::numeric_limits<typename T::underlying>::max();
+  float inv_scale = scale == 0 ? 1.0f : 1.0f / scale;
+  qvalue = lrintf((value - bias) * inv_scale);
```

Comment:
cc @raghuramank100, @dskhudia please check if this equation makes sense. It is what we currently use in caffe2 today for quantizing embedding weights.

Reply:
This looks fine to me. You may also want to check for inv_scale being inf as some day we will hit that case.

Reply:
> You may also want to check for inv_scale being inf as some day we will hit that case.

+1